### PR TITLE
fix: fix id_or_name property of key class

### LINKED ISSUE #106: datastore Key id_or_name() returns None when id is 0
#### Description 
Expecting the Key().is_partial to return False for a Key with id=0.
The is_partial property relies on the id_or_name property.
The id_or_name property returns None when the id=0, which is not expected. Expecting the id to be returned instead.
The id_or_name property implementation is to simple. Should check for None or isinstance(self.id, int)

#### Environment details

  - OS type and version: Ubuntu 20.04
  - Python version: Python 3.8.5
  - pip version: pip 20.2.3
  - `google-cloud-datastore` version: 1.15.3

#### Steps to reproduce

1. Create a datastore key instance with Id = 0
2. Retrieve the id_or_name property, returns None, expected 0.

#### Code example

```python
def test_datastore_key():
    import google.cloud.datastore
    k = google.cloud.datastore.Key('EntityKind', 0, project="unittest")
    assert k.id == 0
    assert k.id_or_name == 0
    assert k.is_partial == False
```

#### Stack trace
```
assert None == 0
  +None
  -0
def test_datastore_key():
        import google.cloud.datastore
        k = google.cloud.datastore.Key('EntityKind', 0, project="unittest")
        assert k.id == 0
>       assert k.id_or_name == 0
E       assert None == 0
E         +None
E         -0
```

### PR DESCRIPTION
**Pull request by HemangChothani**
Fixes #106

### CODE REVIEW

**Comment by chrisrossi** [comment]
Does Datastore let you store a key with id of ``0``?

**Comment by HemangChothani** [comment]
@chrisrossi Library doesn't stop or raise error when passing id =0 but  `if id is  0, generate automatically` as mentioned in document [here](https://cloud.google.com/appengine/docs/standard/go111/datastore/creating-entity-keys#specifying_a_key_name_for_an_entity)

**Comment by chrisrossi** [comment]
> @chrisrossi Library doesn't raise an error when passing id =0 but `if id is 0, generate automatically` as mentioned in document [here](https://cloud.google.com/appengine/docs/standard/go111/datastore/creating-entity-keys#specifying_a_key_name_for_an_entity)

Fair enough! This looks good to me, but I would probably add a regression test.

File: google/cloud/datastore/key.py [hunk 1]
```diff
@@ -441,7 +441,9 @@ def id_or_name(self):
         :returns: The last element of the key's path if it is either an ``id``
                   or a ``name``.
         """
-        return self.id or self.name
+        if self.id is None:
+            return self.name
+        return self.id
```
Comment (tseaver) [inline google/cloud/datastore/key.py:446]: Please add a unit test exercising #106 (i.e., set the key's `id` to 0 and assert that `id_or_name` returns the 0).

**Review by tseaver — CHANGES_REQUESTED** [review]

**Review by tseaver — APPROVED** [review]

**Comment by tseaver** [comment]
@crwilcox, @czahedi Again, the `requiresCodeOwnerReviews: true` policy means my approval is not sufficient to merge this PR.

**Review by crwilcox — APPROVED** [review]

**Pull request merged**